ribbon-eureka/src/main/java/com/netflix/niws/loadbalancer/EurekaNotificationServerListUpdater.java

```diff
@@ -56,12 +58,11 @@ private LazyHolder() {
                             .build()
             );
 
-            poolSizeProp.addCallback(new Runnable() {
+            poolSizeMin.addCallback(new Runnable() {
                 @Override
                 public void run() {
-                    int corePoolSize = getCorePoolSize();
-                    defaultServerListUpdateExecutor.setCorePoolSize(corePoolSize);
-                    defaultServerListUpdateExecutor.setMaximumPoolSize(corePoolSize * 5);
+                    defaultServerListUpdateExecutor.setCorePoolSize(getCorePoolSize());
```

Comment:
FYI, in JDK9, they added more error checking when you call `setCorePoolSize`, so if you attempt to resize higher than the maximum pool size, it'll error out.  We found this within Hystrix as well : https://github.com/Netflix/Hystrix/issues/1874#issuecomment-437005731

Generally, you can just flip them, but then the inverse is true if you try to resize the max lower.  Not sure if we need to accommodate that, but something to be aware of.